### PR TITLE
🐛 Fix mount route positioning

### LINKED ISSUE #2460: 🐛 [Bug]: Wrong handlers execution order in some mount cases
### Bug Description

In some rare cases, fiber can wrongly build `app.treeStack`, breaking the execution chain. For example, it can call a handler from a mounted app before it calls a middleware specified at the top level earlier.

### How to Reproduce

In attached code snippet, fiber thinks that middleware printing `"i'm ignored!"` should be executed after `/methods/test` handler.

I think there are many more cases when this happens, but for now I'm able to reproduce only this one.

### Expected Behavior

Handlers execute in correct order.

### Fiber Version

2.45.0

### Code Snippet (optional)

```go
package main

import (
	"fmt"
	"github.com/gofiber/fiber/v2"
	"net/http"
)

func getApi() *fiber.App {
	var API = fiber.New()

	methods := API.Group("/methods")

	methods.Get("/test", func(c *fiber.Ctx) error {
		if c.Locals("hello") != "world" {
			panic("hello is not world")
		}
		return c.SendStatus(http.StatusOK)
	})

	return API
}

func main() {
	app := fiber.New()

	app.Use(func(c *fiber.Ctx) error {
		return c.Next()
	})

	app.Mount("/ws", fiber.New())

	app.Use(func(c *fiber.Ctx) error {
		return c.Next()
	})

	app.Get("/status", func(c *fiber.Ctx) error {
		return c.SendString("ok")
	})

	app.Use(func(c *fiber.Ctx) error {
		fmt.Println("i'm ignored!")
		c.Locals("hello", "world")
		return c.Next()
	})

	app.Mount("", getApi())

	test(app)
}

func test(app *fiber.App) {
	req, _ := http.NewRequest("GET", fmt.Sprintf("/methods/test"), nil)

	app.Test(req)
}
```

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my problem prior to opening this one.
- [X] I understand that improperly formatted bug reports may be closed without explanation.

### PR DESCRIPTION
**Pull request by ReneWerner87**
## Description

Fixes #2460 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] For new functionalities I follow the inspiration of the express js framework and built them similar in usage
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I tried to make my code as fast as possible with as few allocations as possible

### CODE REVIEW

File: mount_test.go [hunk 1]
```diff
@@ -140,6 +140,75 @@ func Test_App_Mount_Express_Behavior(t *testing.T) {
...
 
+// go test -run Test_App_Mount_RoutePositions
+func Test_App_Mount_RoutePositions(t *testing.T) {
+	t.Parallel()
+	testEndpoint := func(app *App, route, expectedBody string) {
+		resp, err := app.Test(httptest.NewRequest(MethodGet, route, nil))
+		utils.AssertEqual(t, nil, err, "app.Test(req)")
+		body, err := io.ReadAll(resp.Body)
+		utils.AssertEqual(t, nil, err)
+		utils.AssertEqual(t, expectedBody, string(body), "Unexpected response body")
```
Comment (gaby) [inline mount_test.go:151]: Checking status code is missing here

**Review by gaby — APPROVED** [review]
Only one thing missing, but LGTM

File: mount_test.go [hunk 2]
```diff
@@ -140,6 +140,75 @@ func Test_App_Mount_Express_Behavior(t *testing.T) {
 	utils.AssertEqual(t, uint32(16+9), app.routesCount)
 }
 
+// go test -run Test_App_Mount_RoutePositions
+func Test_App_Mount_RoutePositions(t *testing.T) {
+	t.Parallel()
+	testEndpoint := func(app *App, route, expectedBody string) {
+		resp, err := app.Test(httptest.NewRequest(MethodGet, route, nil))
```
Comment (leonklingele) [inline mount_test.go:147]:
```suggestion
		resp, err := app.Test(httptest.NewRequest(MethodGet, route, http.NoBody))
```

This should have been caught by golangci-lint.

**Comment by ReneWerner87** [comment]
@gaby @leonklingele thx for the feedback
can you look again

File: mount_test.go [hunk 3]
```diff
@@ -99,12 +100,13 @@ func Test_App_Mount_Express_Behavior(t *testing.T) {
 			return c.SendString(body)
 		}
 	}
-	testEndpoint := func(app *App, route, expectedBody string) {
-		resp, err := app.Test(httptest.NewRequest(MethodGet, route, nil))
+	testEndpoint := func(app *App, route, expectedBody string, statusCode int) {
```
Comment (leonklingele) [inline mount_test.go:103]:
```suggestion
	testEndpoint := func(app *App, route, expectedBody string, expectedStatusCode int) {
```

**Pull request merged**